server/db/models.js

```diff
@@ -30,15 +30,29 @@ const restaurantSchema = new mongoose.Schema({
 
 const Restaurant = mongoose.model('Restaurant', restaurantSchema);
 
-//  Insert new restaurant into database
+const newRestaurants = (restaurants) => {
+  Restaurant.insertMany(restaurants)
+    .then(data => console.log('documents inserted'))
+    .catch(err => console.log(err));
+};
+
+const getRestaurant = id => Restaurant.find({ id });
+
+const getRestaurantByName = name => Restaurant.find({ name });
+
+const deleteRestaurant = id => Restaurant.deleteOne({ id });
 
-const newRestaurants = (restArr) => {
-  Restaurant.insertMany(restArr).then(data => console.log('documents inserted')).catch(err => console.log(err));
+const updateRestaurant = (id, valuesToUpdate) => {
+  return Restaurant.findOneAndUpdate({ id }, valuesToUpdate, { new: true });
 };
 
-const getRestaurant = (restId, callback) => Restaurant.find({ id: restId }, callback);
+const postRestaurant = newRestaurant => Restaurant.create(newRestaurant);
 
 module.exports = {
   newRests: newRestaurants,
```

Comment:
The shortening of Restaurant to Rest might be confusing for others reading the code base, and afaik variable abbreviation is generally frowned upon in full-stack these days. 

Other than that, looks good to me.

Reply:
Oh and I would add that this one is especially confusing since 'Rest' has a very common meaning itself, especially in Javascript.